client/my-sites/people/role-select/index.jsx

```diff
@@ -15,18 +15,19 @@ import FormLabel from 'calypso/components/forms/form-label';
 import FormRadio from 'calypso/components/forms/form-radio';
 import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
 import QuerySites from 'calypso/components/data/query-sites';
-import QuerySiteRoles from 'calypso/components/data/query-site-roles';
-import { getSite } from 'calypso/state/sites/selectors';
-import { getSiteRoles, getWpcomFollowerRole } from 'calypso/state/site-roles/selectors';
+import { getSite, getWpcomFollowerRole } from 'calypso/state/sites/selectors';
 import { ROLES_LIST } from './constants';
 import isSiteWPForTeams from 'calypso/state/selectors/is-site-wpforteams';
+import useSiteRolesQuery from 'calypso/data/site-roles/use-site-roles-query';
 
 import './style.scss';
 
 const RoleSelect = ( props ) => {
-	let { siteRoles } = props;
+	const { data } = useSiteRolesQuery( props.siteId );
 	const { isWPForTeamsSite } = props;
 
+	let siteRoles = data ?? [];
```

Comment:
This `[]` default changes the rendered UI slightly when `siteRoles` are not loaded yet and the `includeFollower` prop is true:
- **before**: `siteRoles` is null and no radio inputs are rendered
- **after**: `siteRoles` is `[]`, then the follower role is appended and then a list with single radio item is rendered. Only when `siteRoles` response arrives the UI is updated to the full list.

Reply:
Good catch!